framework/src/util/CFRLogger.cpp

```diff
@@ -117,5 +114,32 @@ void CFRLogger::Log(const cppmicroservices::ServiceReferenceBase& sr,
     currLogger->Log(sr, level, message, ex);
   }
 }
+
+void CFRLogger::Open()
+{
+  auto l = this->Lock();
+  US_UNUSED(l);
+  cfrContext = GetBundleContext();
+  serviceTracker.reset(
+    new cppmicroservices::ServiceTracker<cppmicroservices::logservice::LogService> (cfrContext, this));
```

Comment:
prefer using `serviceTracker = std::make_shared<cppmicroservices::cfrimpl::CFRLogger>(...)` instead of `reset()` to avoid calling `new` and leverage better memory management by the c++ runtime.